src/socket_type.cpp

```diff
@@ -88,23 +85,17 @@ namespace aux {
 
 	void setup_ssl_hostname(socket_type& s, std::string const& hostname, error_code& ec)
 	{
-#if defined TORRENT_USE_OPENSSL
+#if defined TORRENT_USE_SSL
 #ifdef TORRENT_MACOS_DEPRECATED_LIBCRYPTO
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #endif
 		// for SSL connections, make sure to authenticate the hostname
 		// of the certificate
 #define CASE(t) case socket_type_int_impl<ssl_stream<t>>::value: \
-		s.get<ssl_stream<t>>()->set_verify_callback( \
-			boost::asio::ssl::rfc2818_verification(hostname), ec); \
-		ssl = s.get<ssl_stream<t>>()->native_handle(); \
-		ctx = SSL_get_SSL_CTX(ssl); \
+		s.get<ssl_stream<t>>()->set_host_name(hostname, ec); \
 		break;
 
-		SSL* ssl = nullptr;
-		SSL_CTX* ctx = nullptr;
-
 		switch(s.type())
```

Comment:
one relatively big change in master is that `socket_type` now behaves more like a `boost::variant`, so instead of these switch statements and saving off SSL pointers, there are visitors. So I think this could be done a bit cleaner in the most recent `master` branch

Reply:
specifically, where the concrete type of the native handles don't necessarily need to be mentioned, but just passed in to free functions that have two implementations, one for openssl and one for GnuTLS

Reply:
I see, that'd be nice. I'm not 100% sure this is doable with free functions for now because of the callbacks but I'll look into it.